app/controllers/instructor/sections_controller.rb

```diff
@@ -36,7 +37,11 @@ def require_authorized_for_current_course
 
 	helper_method :current_course
 	def current_course
-		@current_course ||= Course.find(params[:course_id])
```

Comment:
In this method we say that we should use the `params[:course_id]`, but only if it's passed through the params.  Otherwise, we should use the course that's passed through.